uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/crm/app.form.workpattern.js

```diff
@@ -473,6 +559,53 @@ define([
       }
     };
 
+    /**
+     * Adjusts the time value for the specified type.
+     * It allows to select only times in the specific range:
+     * - "from" time 00:00 to 23:30
+     * - "to" time 00:15 to 23:45
+     *
+     * @param {String} timeType From|To
+     */
+    Day.prototype._adjustTimeValue = function (timeType) {
+      var $input = this._getTimeInputOfType(timeType);
+      var time = $input.val();
+      var momentTime = moment(time, timeFormat);
+      var minTime = moment('00:00', timeFormat);
+      var maxTime = moment('23:45', timeFormat);
+
+      (timeType === 'To') && minTime.add(timeBaseInMinutes, 'minutes');
+      (timeType === 'From') && maxTime.subtract(timeBaseInMinutes, 'minutes');
+
+      if ((!momentTime.isValid() && timeType === 'From') || momentTime.isBefore(minTime)) {
```

Comment:
can we add these conditions to variables?

```
var fromTimeIsBeforeMinTime = (!momentTime.isValid() && timeType === 'From') || momentTime.isBefore(minTime);
var toTimeIsAfterMaxTime = !momentTime.isValid() && timeType === 'To') || momentTime.isAfter(maxTime)

if (fromTimeIsBeforeMinTime) {
  // ...
} else if (toTimeIsAfterMaxTime) {
  // ...
} 
```

Also, can From Time be before Min Time **AND** To Time be after Max Time at the same time? What happens then?

Reply:
Done.

> Also, can From Time be before Min Time AND To Time be after Max Time at the same time? What happens then?

I don't think that's possible.